core/guard/guard.go

```diff
@@ -0,0 +1,117 @@
+package guard
+
+import (
+	"github.com/gogo/protobuf/proto"
+	ic "github.com/libp2p/go-libp2p-core/crypto"
+	"time"
+
+	config "github.com/TRON-US/go-btfs-config"
+	"github.com/TRON-US/go-btfs/core/commands/storage"
+	"github.com/libp2p/go-libp2p-core/peer"
+	"github.com/tron-us/go-btfs-common/crypto"
+	guardPb "github.com/tron-us/go-btfs-common/protos/guard"
+)
+
+func NewContract(session *storage.FileContracts, configuration *config.Config, chunkHash string, chunkIndex int32) (*guardPb.ContractMeta, error) {
+	shard := session.ShardInfo[chunkHash]
+	guardPid, escrowPid, err := getGuardAndEscrowPid(configuration)
+	if err != nil {
+		return nil, err
+	}
+	return &guardPb.ContractMeta{
+		ContractId:       shard.ContractID,
+		RenterPid:        session.Renter.Pretty(),
+		HostPid:          shard.Receiver.Pretty(),
+		ShardHash:        chunkHash,
+		ShardIndex:       chunkIndex,
+		FileHash:         session.FileHash.KeyString(),
+		RentStart:        shard.StartTime,
+		RentEnd:          shard.StartTime.Add(shard.Length),
+		GuardPid:         guardPid.Pretty(),
+		EscrowPid:        escrowPid.Pretty(),
+		Price:            shard.Price,
+		Amount:           1000,
```

Comment:
Where did amount come from? is this from config or passed in?

Reply:
I think that need to be calculated. Price is per unit time and per unit size? That is determined by later policy so I set a default value here.

Reply:
The policy is already here, please check:
```
  int64 price = 12; //the price which agreed by host and rental, on the first phase, it equal to the proposed_price
  int64 amount = 13; //the amount which agreed by host and rental, on the first phase, it should be equal file_size*duration*agreed_price
```

Reply:
Please implement, otherwise the amount is wrong

Reply:
how can I get file size? that should be shard size right? Then is this const in somewhere? And what's the default length

Reply:
FileSize is the size of the shard, it's available in ContractMeta https://github.com/TRON-US/go-btfs-common/blob/master/protos/guard/guard.proto#L50

Duration is the passed in parameter we talked about before in BTFS-1101, and it can also be calculated from `rent_end-rent_start`. Now I am confused, what is `shard.Length`?? https://github.com/TRON-US/go-btfs/pull/271/files#diff-1e19489fd20fa88c4906b73e5b0d240cR30

Reply:
length should be rent_end-rent_start, I set a default value here.
file size is just what I am going to pass in. But I don't know how big is that.

Reply:
I don't see where you set the default, it's always unset (0). But that's fine, you will set it later.

For now you should use `shard.Length` for duration computation.

For file size, you need to `api.ResolveNode` on the cid, then use `Size()` on that node to obtain size.

Reply:
I will add setting length as option

Reply:
can you call it `contract-length`? to make it easier to understand